packages/dds/tree/src/core/revertible.ts

```diff
@@ -36,6 +38,26 @@ export interface Revertible {
 	dispose(): void;
 }
 
+/**
+ * Provides clone functionality for {@link Revertible}.
+ *
+ * @alpha
+ */
+export interface RevertibleAlpha extends Revertible {
+	/**
+	 * TODO: Add description
+	 *
+	 * @param forkedBranch - A target branch to apply the revertible to. If not given, spawns a cloned revertible of the original branch.
+	 * @returns A cloned revertible.
+	 */
+	clone: (forkedBranch?: TreeBranch) => RevertibleAlpha;
+}
+
+/**
+ * Union type for {@link Revertible} and {@link RevertibleAlpha}.
+ */
+export type RevertibleType = Revertible | RevertibleAlpha;
```

Comment:
Is this type ever useful? `RevertibleAlpha` extends `Revertible`, therefore anywhere that `RevertibleType` is used, `RevertibleAlpha` can be used directly instead.

An analogy would be the type union of `Cat | Animal` - it can always be replaced with just `Animal`.

Reply:
NUBE MISTAKE Alert 😢 . Thanks for pointing this out!